server/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -718,6 +733,40 @@ public boolean isRequireAlias() {
         return requireAlias;
     }
 
+    @Override
+    public Index getConcreteWriteIndex(IndexAbstraction ia, Metadata metadata) {
```

Comment:
It feels like a bit too much logic for a data class, that the IndexRequest is essentially is. I wonder if it makes more sense as a part of IndexAbstraction instead.

Reply:
Implemented via: f97d88935098460c2f1a7cfafb6d4271636b60a5